presto-main/src/main/java/com/facebook/presto/operator/project/InputPageProjection.java

```diff
@@ -56,32 +57,44 @@ public InputChannels getInputChannels()
     }
 
     @Override
-    public PageProjectionOutput project(ConnectorSession session, DriverYieldSignal yieldSignal, Page page, SelectedPositions selectedPositions)
+    public Work<Block> project(ConnectorSession session, DriverYieldSignal yieldSignal, Page page, SelectedPositions selectedPositions)
     {
-        return new InputPageProjectionOutput(page, selectedPositions);
+        return new InputPageProjectionWork(page, selectedPositions);
     }
 
-    private class InputPageProjectionOutput
-            implements PageProjectionOutput
+    private class InputPageProjectionWork
+            implements Work<Block>
     {
         private final Block block;
         private final SelectedPositions selectedPositions;
+        private Block result;
 
-        public InputPageProjectionOutput(Page page, SelectedPositions selectedPositions)
+        public InputPageProjectionWork(Page page, SelectedPositions selectedPositions)
         {
             this.block = requireNonNull(page, "page is null").getBlock(0);
             this.selectedPositions = requireNonNull(selectedPositions, "selectedPositions is null");
         }
 
         @Override
-        public Optional<Block> compute()
+        public boolean process()
         {
+            verify(result == null, "result has been generated");
```

Comment:
`checkState`. Check other usages of `verify` in this pull request.

Here, `result == null` would be a violation of calling contract. It could happen if used incorrectly. This is a perfect fit for `IllegalStateException`.